aspnetcore/performance/timeouts/samples/8.x/Program.cs

```diff
@@ -0,0 +1,157 @@
+});
+// </definepolicies>
+
+var app = builder.Build();
+app.UseRequestTimeouts();
+
+// <usepolicy>
+app.MapGet("/namedpolicy", async (HttpContext context) => {
+    await Task.Delay(TimeSpan.FromSeconds(2));
+
+    if (context.RequestAborted.IsCancellationRequested) {
+        return Results.Content("Timeout!", "text/plain");
+    }
+
+    return Results.Content("No timeout!", "text/plain");
+}).WithRequestTimeout("MyPolicy");
+// Returns "Timeout!"
+// </usepolicy>
+
+// <usedefault>
+app.MapGet("/", async (HttpContext context) => {
+    await Task.Delay(TimeSpan.FromSeconds(2));
+
+    if (context.RequestAborted.IsCancellationRequested) {
+        return Results.Content("Timeout!", "text/plain");
+    }
+
+    return Results.Content("No timeout!", "text/plain");
+});
+// Returns "Timeout!" due to default policy.
+// </usedefault>
+
+// <disableall>
+app.MapGet("/disableall", [DisableRequestTimeout] async (HttpContext context) => {
+    await Task.Delay(TimeSpan.FromSeconds(2));
+
+    return Results.Content("No timeout!", "text/plain");
+});
+// Returns "No timeout!", ignores default timeout.
+// </disableall>
+app.Run();
+#endif
+
+#if policies2
+using Microsoft.AspNetCore.Http.Timeouts;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// <definepolicies2>
+builder.Services.AddRequestTimeouts(options => {
+    options.DefaultPolicy = new RequestTimeoutPolicy {
+        Timeout = TimeSpan.FromMilliseconds(1000),
+        TimeoutStatusCode = 504
+    };
+    options.AddPolicy("MyPolicy2", new RequestTimeoutPolicy {
+        Timeout = TimeSpan.FromMilliseconds(1000),
+        WriteTimeoutResponse = async (HttpContext context) => {
+            context.Response.ContentType = "text/plain";
+            await context.Response.WriteAsync("Timeout!");
+        }
+    });
+});
+// </definepolicies2>
+
+var app = builder.Build();
+app.UseRequestTimeouts();
+
+// <usedefault2>
+app.MapGet("/", async (HttpContext context) => {
+    await Task.Delay(TimeSpan.FromSeconds(2));
+
+    context.RequestAborted.ThrowIfCancellationRequested();
+
+    return Results.Content("No timeout!", "text/plain");
+});
+// Returns status code 504 due to default policy.
+// </usedefault2>
+
+// <usepolicy2>
+app.MapGet("/usepolicy2", async (HttpContext context) => { 
+    await Task.Delay(TimeSpan.FromSeconds(2));
+
+    context.RequestAborted.ThrowIfCancellationRequested();
+
+    return Results.Content("No timeout!", "text/plain");
+}).WithRequestTimeout("MyPolicy2");
+// Returns "Timeout!" due to WriteTimeoutResponse in MyPolicy2.
+// </usepolicy2>
+
+// <canceltimeout>
+app.MapGet("/canceltimeout", async (HttpContext context) => {
+    await Task.Delay(TimeSpan.FromSeconds(2));
+
+    if (context.RequestAborted.IsCancellationRequested)
+    {
+        var timeoutFeature = context.Features.Get<IHttpRequestTimeoutFeature>();
+        timeoutFeature?.DisableTimeout();
+    }
+
+    return Results.Content("No timeout!", "text/plain");
+});
```

Comment:
I think this sample shows better that how `DisableTimeout` would work.

```suggestion
app.MapGet("/canceltimeout", async (HttpContext context) => {

    var timeoutFeature = context.Features.Get<IHttpRequestTimeoutFeature>();
    timeoutFeature?.DisableTimeout();

    await Task.Delay(TimeSpan.FromSeconds(2));
    
    if (context.RequestAborted.IsCancellationRequested) // is false
    {

    }

    return Results.Content("No timeout!", "text/plain");
}).WithRequestTimeout(TimeSpan.FromSeconds(1));
```